starter-code/main.js

```diff
@@ -53,7 +53,182 @@ mongoClient.connect(`mongodb://localhost:27017/crunchbase`, (error, db) => {
             })
             break;
 
-          // Code here next cases!
+            case "4":
+              // 4.- List by name all companies founded in february of 2004.
+            db.collection('companies').find({ $and: [{ founded_year: 2004 }, { founded_month: 2 }] }, { name: 1, _id: 0 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+            case "5":
+              // 5.- List by name all companies founded in the summer of 2004 (april to june) sorted by date.
+            db.collection('companies').find({ $and: [{ founded_year: 2004 }, { founded_month: { $gte: 4 } }, { founded_month: { $lte: 6 } }]}, { name: 1, _id: 0 }).sort({ founded_month: 1, founded_day: 1}).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break
+
+            case "6":
+              // 6.- What companies have offices in "Barcelona".
+            db.collection('companies').find({ "offices.city": "Barcelona" }, { name: 1, _id: 0 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+            case "7":
+             //  7.- List the 10 companies with more employees sorted ascending (show name and employees).
+
+            db.collection('companies').find({ name: 1, _id: 0 }).sort({number_of_employees: -1}).limit(10).toArray((error, result) => {
```

Comment:
El primer argumento es siempre una búsqueda no una proyección. Si no quieres especificar criterios usa una query vacía.
```suggestion
            db.collection('companies').find({}, { name: 1, _id: 0 }).sort({number_of_employees: -1}).limit(10).toArray((error, result) => {
```